app/controllers/LongerCommodityCodeController.scala

```diff
@@ -52,79 +51,74 @@ class LongerCommodityCodeController @Inject() (
     extends FrontendBaseController
     with I18nSupport {
 
-  private val form = formProvider()
+  private val form           = formProvider()
+  private val validLength    = 6
+  private val maxValidLength = 10
 
   def onPageLoad(mode: Mode, recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData) {
     implicit request =>
-      val categorisationInfoOpt: Option[CategorisationInfo] =
-        request.userAnswers.get(RecordCategorisationsQuery).flatMap(_.records.get(recordId))
-
-      val originalComcodeOpt = categorisationInfoOpt.flatMap(_.originalCommodityCode)
-      val latestComcodeOpt   = categorisationInfoOpt.map(_.commodityCode)
-
-      val answerToFillFormWith = for {
-        originalCode <- originalComcodeOpt
-        latestCode   <- latestComcodeOpt
-      } yield latestCode.drop(originalCode.length)
-
-      val preparedForm = (categorisationInfoOpt, answerToFillFormWith) match {
-        case (Some(categorisationInfo), Some(answerToFillFormWith)) if categorisationInfo.latestDoesNotMatchOriginal =>
-          form.fill(answerToFillFormWith)
-        case _                                                                                                       =>
-          form
-      }
-
-      originalComcodeOpt match {
-        case Some(shortCommodity) if commodityCodeSansTrailingZeros(shortCommodity).length == 6 =>
-          Ok(view(preparedForm, mode, commodityCodeSansTrailingZeros(shortCommodity), recordId))
-        case _                                                                                  =>
+      val previousAnswerOpt = request.userAnswers.get(LongerCommodityCodePage(recordId))
+      val shortComcodeOpt   = getShortCommodityCodeOpt(recordId, request, validLength)
+      val preparedForm      = previousAnswerOpt.map(answer => form.fill(answer)).getOrElse(form)
```

Comment:
Can we use this pattern so it matches other pages

`      val preparedForm = request.userAnswers.get(LongerCommodityCodePage(recordId)) match {
        case None        => form
        case Some(value) => form.fill(value)
      }`

Reply:
Sure :)